### PR TITLE
use dedicated query node for tx and query RPC

### PR DESCRIPTION
**Pull request by MSalopek**
Add RPC node to integration test configs.

Each chain now has a defined validator set and a dedicated non-validating full node used for tx and query broadcast.

Closes: #263

### CODE REVIEW

**Review by sainoe — APPROVED** [review]
LGTM. Great work @MSalopek

File: tests/integration/state.go [hunk 1]
```diff
@@ -476,3 +473,19 @@ func (tr TestRun) getValidatorIP(chain chainID, validator validatorID) string {
...
+	return fmt.Sprintf("tcp://%s:26658", tr.getQueryNodeIP(chain))
+}
+
+// getQueryNodeIP returns query node IP for chain,
+// ipSuffix is hardcoded to be 253 on all query nodes.
+func (tr TestRun) getQueryNodeIP(chain chainID) string {
+	return fmt.Sprintf("%s.253", tr.chainConfigs[chain].ipPrefix)
+}
+
+// getQueryNodeHome returns query node home on chain.
```
Comment (smarshall-spitzbart) [inline tests/integration/state.go:488]: Is this needed?

File: tests/integration/testnet-scripts/start-chain.sh [hunk 2]
```diff
@@ -230,12 +230,65 @@ do
     ip netns exec $NET_NAMESPACE_NAME $BIN $ARGS start &> /$CHAIN_ID/validator$VAL_ID/logs &
 done
 
+## ADD QUERY NODE
```
Comment (smarshall-spitzbart) [inline tests/integration/testnet-scripts/start-chain.sh:233]:
```suggestion
## SETUP QUERY NODE NETWORK NAMESPACE
```

File: tests/integration/testnet-scripts/start-chain.sh [hunk 3]
```diff
@@ -230,12 +230,65 @@ do
...
+QUERY_IP_SUFFIX="253"
+QUERY_NET_NAMESPACE_NAME="$CHAIN_ID-$QUERY_NODE_ID"
+QUERY_IP_ADDR="$CHAIN_IP_PREFIX.$QUERY_IP_SUFFIX/24"
+
+ip netns add $QUERY_NET_NAMESPACE_NAME
+ip link add $QUERY_NET_NAMESPACE_NAME-in type veth peer name $QUERY_NET_NAMESPACE_NAME-out
+ip link set $QUERY_NET_NAMESPACE_NAME-in netns $QUERY_NET_NAMESPACE_NAME
+ip netns exec $QUERY_NET_NAMESPACE_NAME ip addr add $QUERY_IP_ADDR dev $QUERY_NET_NAMESPACE_NAME-in
+ip link set $QUERY_NET_NAMESPACE_NAME-out master virtual-bridge
+## DONE ADD QUERY NODE
```
Comment (smarshall-spitzbart) [inline tests/integration/testnet-scripts/start-chain.sh:244]:
```suggestion
## DONE SETUP QUERY NODE NETWORK NAMESPACE
```

**Review by smarshall-spitzbart — APPROVED** [review]
Just a few small things, nice job!

**Review by danwt — APPROVED** [review]
Nice work @MSalopek!

File: tests/integration/state.go [hunk 4]
```diff
@@ -440,16 +440,13 @@ func (tr TestRun) getParam(chain chainID, param Param) string {
 	return value.String()
 }
 
-// Gets a default validator for txs and queries using the first subdirectory
-// of the directory of the input chain, which will be the home directory
-// of one of the validators.
-// TODO: Best solution for default validator fulfilling queries etc. is a dedicated, non validating, full node.
-// See https://github.com/cosmos/interchain-security/issues/263
+// Gets a default validator for input chain by parsing node home directories.
+//lint:ignore U1000 Ignore unused function temporarily for debugging
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply tests/integration/state.go:444]: Removed.

`Done.`

File: tests/integration/state.go (hunk 1 above)
Comment (MSalopek) [reply tests/integration/state.go:488]: Removed.

`Done.`

File: tests/integration/testnet-scripts/start-chain.sh (hunk 2 above)
Comment (MSalopek) [reply tests/integration/testnet-scripts/start-chain.sh:233]: `Done.`

File: tests/integration/testnet-scripts/start-chain.sh (hunk 3 above)
Comment (MSalopek) [reply tests/integration/testnet-scripts/start-chain.sh:244]: `Done.`

**Pull request merged**